pkg/controller/controller.go

```diff
@@ -102,8 +115,47 @@ func NewController(clientset kubernetes.Interface, ssclientset ssclientset.Inter
 		},
 	})
 
+	sInformer := sinformer.Core().V1().Secrets().Informer()
+	sInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: func(obj interface{}) {
+			skey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			if err != nil {
+				log.Printf("failed to fetch Secret key: %v", err)
+				return
+			}
+
+
+			if (!findOwnerReferenceSS(obj)) {
+				if (!isAnnotatedToBeManaged(obj.(*corev1.Secret))) {
```

Comment:
This could be a simple && expression right?

Not a big deal anyways... I am not fan on nesting I guess, specially when not needed.

Reply:
totally :')